### DISCUSSION TITLE
How to `read` an escape character?

### DISCUSSION THREAD

**Discussion by elder-n00b** [fish-shell/fish-shell#8711] [Q&A]
In bash, I can `read -n1 myvar` and if ESC is pressed `$var` contains the escape: `printf "%q" "$myvar" # ==> $'\E'`.
If fish, pressing ESC doesn't seem to have any effect. Nevermind there's no `printf %q` either.
It's obviously convenient in prompts, of course I can use Y/N, but sometimes ENTER/ESC is just more intuitive and faster to type.
I don't see anything about it in the [tutorial](https://fishshell.com/docs/current/tutorial.html), the [docs](https://fishshell.com/docs/current/cmds/read.html?highlight=read), nor `man read`.
Am I missing something? Is there a different way?

Unrelated nitpicking:
`bash`: `read ... [-p  prompt] ...` (a prompt **string**)
`fish`: `read ... [-p PROMPT_CMD] [-P PROMPT_STR] [-R RIGHT_PROMPT_CMD] ...`

Oh well, just one more thing to remember I guess... hopefully I'll never `read -p 'rm -rf something Y/N'`.

**Comment by zanchey** [comment]
> Unrelated nitpicking:
> `bash`: `read ... [-p prompt] ...` (a prompt **string**)
> `fish`: `read ... [-p PROMPT_CMD] [-P PROMPT_STR] [-R RIGHT_PROMPT_CMD] ...`

Indeed, `--silent` and `--shell` used to be different to bash too. It was changed in 3.0. I'd think about doing the same for `-p`/`-P` except that it's got a lot of potential to blow up (because strings will be executed as fish script).
